coalib/settings/Setting.py

```diff
@@ -296,3 +297,26 @@ def value(self):
                              'incomplete. Please access the value of the '
                              'setting in a section to get the complete value.')
         return self._value
+
+    @property
+    def origin(self):
+        """
+        Ensures that Setting.origin always returns a string,
```

Comment:
@li-boxuan  But i will call it as `Setting.origin` only. What would be good docstring otherwise?

Reply:
@akshatkarani you shouldnt mention `Setting.origin` here, as it is the docstring of `Setting.origin`.  and instead of `string...`, just say `filename`